django/contrib/postgres/aggregates/general.py

```diff
@@ -100,4 +102,5 @@ class StringAgg(DeprecatedConvertValueMixin, OrderableAggMixin, Aggregate):
 
     def __init__(self, expression, delimiter, **extra):
         delimiter_expr = Value(str(delimiter))
+        extra['output_field'] = TextField() if extra.get('output_field') is None else extra['output_field']
```

Comment:
You can define the `output_field` as a class attribute instead to avoid this boilerplate. See how it's done for `JSONBAgg` just above for example.